systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java

```diff
@@ -437,6 +447,185 @@ void testCreateTopicAfterUnsupportedOperation(ExtensionContext extensionContext)
+
+        reason = "ReplicationFactorChangeException";
+        reasonMessage = "Changing 'spec.replicas' is not supported.";
+        KafkaTopicUtils.waitForKafkaTopicNotReady(namespace, topicName);
+        assertMetricResourceState(toMetricsCollector, KafkaTopic.RESOURCE_KIND, topicName, namespace, 0, reasonMessage);
+        assertKafkaTopicStatus(topicName, namespace, NotReady, reason, reasonMessage, 3);
+
+        LOGGER.info("Changing KafkaTopic's spec to correct state");
+        KafkaTopicResource.replaceTopicResourceInSpecificNamespace(topicName, kafkaTopic -> kafkaTopic.getSpec().setReplicas(initialReplicas), namespace);
+        KafkaTopicUtils.waitForKafkaTopicReady(namespace, topicName);
+        assertKafkaTopicStatus(topicName, namespace, Ready, 4);
+
+        reasonMessage = "none";
+        assertMetricResourceState(toMetricsCollector, KafkaTopic.RESOURCE_KIND, topicName, namespace, 1, reasonMessage);
+        KafkaTopicUtils.waitForKafkaTopicReady(namespace, topicName);
+
+        LOGGER.info("Decreasing number of partitions to {}", decreasePartitions);
+        KafkaTopicResource.replaceTopicResourceInSpecificNamespace(topicName, kafkaTopic -> kafkaTopic.getSpec().setPartitions(decreasePartitions), namespace);
+        KafkaTopicUtils.waitForKafkaTopicPartitionChange(namespace, topicName, decreasePartitions);
+        KafkaTopicUtils.waitForKafkaTopicNotReady(namespace, topicName);
+
+        reason = "PartitionDecreaseException";
+        reasonMessage = "Number of partitions cannot be decreased";
+        assertKafkaTopicStatus(topicName, namespace, NotReady, reason, reasonMessage, 5);
+
+        // Wait some time to check if error is still present in KafkaTopic status
+        LOGGER.info("Wait {} ms for next reconciliation", topicOperatorReconciliationInterval);
+        Thread.sleep(topicOperatorReconciliationInterval);
+        assertKafkaTopicStatus(topicName, namespace, NotReady, reason, reasonMessage, 5);
+
+        LOGGER.info("Changing KafkaTopic's spec to correct state");
+        KafkaTopicResource.replaceTopicResourceInSpecificNamespace(topicName, kafkaTopic -> {
+            kafkaTopic.getSpec().setReplicas(initialReplicas);
+            kafkaTopic.getSpec().setPartitions(initialPartitions);
+        }, namespace);
+        KafkaTopicUtils.waitForKafkaTopicReady(namespace, topicName);
+        assertKafkaTopicStatus(topicName, namespace, Ready, 6);
+
+        // Check failed reconciliations
+        // This is currently doesn't work properly and will be changed in UTO implementation - https://github.com/strimzi/proposals/pull/76
+        // assertMetricValueHigherThan(toMetricsCollector, "strimzi_reconciliations_failed_total{kind=\"" + KafkaTopic.RESOURCE_KIND + ",}", 3);
+    }
+
+    @ParallelTest
+    @KRaftNotSupported("TopicOperator is not supported by KRaft mode and is used in this test class")
+    // The test should be merged with testKafkaTopicDifferentStates when the issue wull be fixed
```

Comment:
```suggestion
    // The test should be merged with testKafkaTopicDifferentStates when the issue will be fixed
```

Reply:
Out of curiosity, did you try that it still doesn't work? IIRC we had a conversation around that. If not, we can test it afterwards

Reply:
This one is connected to the issue which isn't resolved and I think we discussed that this particular issue still exists. I can try it on Monday if it works.

Reply:
Testes are working fine, removing disabled annotation